src/ixdat/techniques/ec_ms.py

```diff
@@ -426,6 +429,156 @@ def siq_ecms_calibration_curve(
+                (signal_response[1], np.zeros(len(v_sig) - len(signal_response[1])))
+            )
+            v_sig_corr = v_sig
+        else:
+            kernel = signal_response[1]
+            v_sig_corr = np.hstack(
+                (v_sig, np.zeros(len(signal_response[1]) - len(v_sig)))
+            )
+        H = fft(kernel)
+        # TODO: store this as well.
+        partial_current = np.real(
+            ifft(fft(v_sig_corr) * np.conj(H) / (H * np.conj(H) + (1 / snr) ** 2))
+        )
+        partial_current = partial_current * sum(kernel)  # what does this do????
+        if len(t_sig) < len(partial_current):
+            delta = len(partial_current) - len(t_sig)
+            partial_current = partial_current[:-delta]
+        return t_sig, partial_current
+
+    def extract_impulse_response(self, mol, tspan=None, tspan_bg=None):
+        """Extracts an ECMSImpulseResponse object from a measurement using the
+        algorithm developed by Krempl et al.
+        https://pubs.acs.org/doi/abs/10.1021/acs.analchem.1c00110
+
+        # TODO: add some option of plotting the impulse response together with the data
+        # TODO: re-add a possibility to choose cut-off potentials rather
+        than having to specify the tspan to make the existance of this method more
+        meaningful, right now it just returns an array of the calibrated signal of mol
+        normalized to the peak area
+        Args:
+            mol (str): Molecule from which the impulse response is to be extracted.
+            tspan(list): Timespan over which which the impulse response is
+                extracted.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+        """
+        # grab the calibrated data
+        impulse_response = ECMSImpulseResponse(
+            mol=mol, data=self
+        ).calc_impulse_response_from_data(tspan=tspan, tspan_bg=tspan_bg)
+        return impulse_response
+
+    def deconvolute_for_tspans(
+        self,
+        tspan_list,
+        t_zero_list,
+        impulse_response,
+        mol,
+        F_mol,
+        plot=True,
+        name=None,
+        t_bg=[-10, -1],
+        snr=7,
+        return_t_v_list=False,
+    ):
+        """
+        Loops though list of tspans and associated t_zero list to deconvolute using
+        the given impulse response (from model, but could also be from data) for
+        the molecule the impulse resp
+
+        The plots will only be meaningful if the current is calibrated.
+        #TODO: add error message
+        Args:
+            tspan_list (list): list of tspans to devonvolute data over
+            t_zero_list (list): zero point where the deconvolution starts
+                #TODO:need to understand if this is actually necessary
+            impulse_response (ImpulseResponse): impulse response object from model/data
+            mol (str): molecule
+            F_mol (CalPoint): spectro_inlets_calibration CalPoint object
```

Comment:
The `data_snippet` should inherit all of the original measurement's `Calculator`s including siq *or* native MS calculators, so you should be able to drop this argument and instead just require that the original measurement has a Calculator for `mol` and use `grab_flux`.